### PR TITLE
Fixes #8154: Printf error messages on Windows Agents

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/8154

### CODE REVIEW

File: tree/20_cfe_basics/time_lib.cf [hunk 1]
```diff
@@ -68,8 +68,12 @@ bundle agent ncf_splay(test_splay)
       "splay_max" string => "268435455";
       # pseudorandon deterministic hex number of 7 nibbles
       "splay_hex" string => string_head(hash("${sys.host}${sys.ipv4}", "md5"), "7");
+
       # decimal version of the splay integer
+    !windows:
       "splay" string => execresult("/usr/bin/printf '%d' 0x${splay_hex}", "noshell");
+    windows::
+      "splay" string => execresult(" [Convert]::ToInt32('0x${splay_hex}', 16)", "powershell");
```
Comment (amousset) [inline tree/20_cfe_basics/time_lib.cf]: Is the space at the beginning of the line needed?

**Comment by ncharles** [comment]
Commit modified

File: tree/20_cfe_basics/time_lib.cf (hunk 1 above)
Comment (ncharles) [inline tree/20_cfe_basics/time_lib.cf]: this is clearly a typo :/

File: tree/20_cfe_basics/time_lib.cf [hunk 2]
```diff
@@ -68,8 +68,12 @@ bundle agent ncf_splay(test_splay)
       "splay_max" string => "268435455";
       # pseudorandon deterministic hex number of 7 nibbles
       "splay_hex" string => string_head(hash("${sys.host}${sys.ipv4}", "md5"), "7");
+
       # decimal version of the splay integer
+    !windows:
```
Comment (amousset) [inline tree/20_cfe_basics/time_lib.cf]: `!windows:` -> `!windows::`

Comment (ncharles) [inline tree/20_cfe_basics/time_lib.cf]: ok, this PR is clearly not my best one ...

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**